File: MEIv3/app/src/main/java/com/example/ramir/meiv3/login_fragment.java
package com.example.ramir.meiv3;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.ConsoleMessage;
import android.webkit.JavascriptInterface;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.squareup.picasso.Picasso;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

/*
 * Created by ramir on 19/03/2017.
 */

public class login_fragment extends Fragment {
    private static final String ARG_SECTION_NUMBER = "section_number";
    private String TAG = "Login Activity";
    private String PagMadre = new GlobalVars().urlMEIMaster()+"login.php";
    private WebView MEIPage;
    private ProgressBar barra;
    private EditText user,pass;
    private LinearLayout linearLayout;
    private ImageView logo;

    public static login_fragment newInstance(int sectionNumber){
        login_fragment fragment = new login_fragment();
        Bundle args = new Bundle();
        args.putInt(ARG_SECTION_NUMBER, sectionNumber);
        fragment.setArguments(args);
        return fragment;
    }

    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, Bundle savedInstanceState){
        View rootView = inflater.inflate(R.layout.fragment_login,container,false);

        MEIPage = (WebView) rootView.findViewById(R.id.webview);
        barra = (ProgressBar) rootView.findViewById(R.id.progressBar);
        user = (EditText) rootView.findViewById(R.id.et_email);
        pass = (EditText) rootView.findViewById(R.id.et_pass); //Declaraciones
        Button login = (Button) rootView.findViewById(R.id.btn_login);
        linearLayout = (LinearLayout) rootView.findViewById(R.id.til);
        logo = (ImageView) rootView.findViewById(R.id.logoView);

        getActivity().runOnUiThread(new Runnable() {
            @Override
            public void run() {
                Picasso.with(getContext()).load(PagMadre+"/../imgs/logo1.png").into(logo);
            }
        });

        WebSettings webSettings = MEIPage.getSettings();
        webSettings.setJavaScriptEnabled(true); //Settings de MEIPage
        webSettings.setDomStorageEnabled(true);

        MEIPage.addJavascriptInterface(new JavaScriptInterface(getContext()), "KOTLIN");

        MEIPage.setWebViewClient(new WebViewClient() {//Cliente para funciones del WebView

            @Override
            public void onPageStarted(WebView view, String url, Bitmap favicon) { //Cuando la pagina empieza a cargar
                barra.setVisibility(View.VISIBLE); //La barra se muestra
            }

            @Override
            public void onPageFinished(final WebView view, String url) { //Cuando la pagina se cargo ejecuta esto
                barra.setVisibility(View.GONE); //La barra se esconde
                linearLayout.setVisibility(View.VISIBLE);
            }

        });
        MEIPage.setWebChromeClient(new WebChromeClient(){ //ChromeClient para leer la consola de JS

            public boolean onConsoleMessage(ConsoleMessage cm) {
                Log.i(TAG, cm.message());
                return true;
            }

        });

        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final String email = user.getText().toString();
                final String passw = pass.getText().toString();
                Boolean error=true;
                if (!isValidEmail(email)) {
                    user.setError("Correo electrónico inválido.");
                    error = false;
                }
                if (!isValidPassword(passw)) {
                    pass.setError("Contraseña inválida.");
                    error = false;
                }

                if(error){
                    MEIPage.loadUrl("javascript:" +
                            "var usr = document.getElementById('input_usuario');" +
                            "var pass = document.getElementById('input_pass');" +
                            "usr.value='" + user.getText().toString() + "';" +
                            "pass.value='" + pass.getText().toString() + "';" +
                            "$('#btn_submit').click();");
                }
            }
        });

        MEIPage.loadUrl(PagMadre);
        return rootView;
    }

    private class JavaScriptInterface{
        Context context;

        JavaScriptInterface(Context c) {
            context = c;
        }

        @JavascriptInterface
        public void pageLoaded(int i){
            if(i == 2){
                getActivity().runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        user.setText("");
                        pass.setText("");
                        startActivity(new Intent(getActivity(), SesionActivity.class));
                        getActivity().finish();
                    }
                });
            }
        }

        @JavascriptInterface
        public void loginError(){
            Toast.makeText(getContext(),"Correo electrónico o contraseña incorrectos, por favor, intentelo de nuevo.", Toast.LENGTH_SHORT).show();
            pass.setText("");
        }
    }

    private boolean isValidPassword(String pass) {
        return pass != null && pass.length() > 6;
    }

    private boolean isValidEmail(String email) {
        String EMAIL_PATTERN = getResources().getString(R.string.valid_email_pattern);

        Pattern pattern = Pattern.compile(EMAIL_PATTERN);
        Matcher matcher = pattern.matcher(email);
        return matcher.matches();
    }
}

File: MEINew/app/src/main/java/ramir/mei/LoginFragment.kt
package ramir.mei

import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.*
import android.widget.Toast
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.fragment_login.*
import kotlinx.android.synthetic.main.fragment_login.view.*
import java.util.regex.Pattern

class LoginFragment : android.app.Fragment() {
    private var pagMEI = ""
    private val mJSIName : String = GlobalVar().getJSIName()
    private val mTAG = "Login Activity"

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater!!.inflate(R.layout.fragment_login, container, false)
        pagMEI = GlobalVar().getMeiURL(activity) + "login.php"

        activity.runOnUiThread { Picasso.with(activity.baseContext).load(pagMEI + "/../imgs/logo1.png").into(rootView.logoView) }

        val MEI = WebView(activity)

        val webSettings = MEI.settings
        webSettings.javaScriptEnabled = true
        webSettings.domStorageEnabled = true

        MEI.addJavascriptInterface(JavaScriptInterface(activity.baseContext), mJSIName)

        MEI.webViewClient = object : WebViewClient() {

            override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
                super.onPageStarted(view, url, favicon)
                rootView.progressBar.visibility = View.VISIBLE
            }

            override fun onPageFinished(view: WebView, url: String) {
                rootView.progressBar.visibility = View.GONE
                rootView.scrollLogin.visibility = View.VISIBLE
            }

        }
        MEI.webChromeClient = object : WebChromeClient() {
            override fun onConsoleMessage(cm: ConsoleMessage): Boolean {
                Log.i(mTAG, cm.message())
                return true
            }
        }

        rootView.btn_login.setOnClickListener {
            val email = rootView.et_email.text.toString()
            val passw = rootView.et_pass.text.toString()

            var error: Boolean? = true

            if (!isValidEmail(email)) {
                rootView.et_email.error = "Correo electrónico inválido."
                error = false
            }

            if (!isValidPassword(passw)) {
                rootView.et_pass.error = "Contraseña inválida."
                error = false
            }

            if (error!!) {
                MEI.loadUrl("javascript:" +
                        "var usr = document.getElementById('input_usuario');" +
                        "var pass = document.getElementById('input_pass');" +
                        "usr.value='" + et_email.text.toString() + "';" +
                        "pass.value='" + et_pass.text.toString() + "';" +
                        "$('#btn_submit').click();")
            }
        }

        MEI.loadUrl(pagMEI)
        return rootView
    }



    private inner class JavaScriptInterface internal constructor(internal var context: Context) {

        @JavascriptInterface
        fun pageLoaded(i: Int) {
            if(i == 2){
                activity.runOnUiThread {
                    et_email.setText("")
                    et_pass.setText("")
                    startActivity(Intent(activity, LoggedFragment::class.java))
                    activity.finish()
                }
            }
        }

        @JavascriptInterface
        fun loginError() {
            Toast.makeText(context, "Correo electrónico o contraseña incorrectos, por favor, intentelo de nuevo.", Toast.LENGTH_SHORT).show()
            et_pass.setText("")
        }
    }

    private fun isValidPassword(pass: String?): Boolean {
        return pass != null && pass.length > 6
    }

    private fun isValidEmail(email: String): Boolean {
        val regex = resources.getString(R.string.valid_email_pattern)

        val pattern = Pattern.compile(regex)
        val matcher = pattern.matcher(email)
        return matcher.matches()
    }
}
